pages/src/rtl-switcher/rtl-switcher.ts

```diff
@@ -0,0 +1,81 @@
+import {ready} from '../../../src/modules/esl-utils/decorators/ready';
+import {attr} from '../../../src/modules/esl-base-element/decorators/attr';
+import {bind} from '../../../src/modules/esl-utils/decorators/bind';
+import {boolAttr, ESLBaseElement} from '../../../src/modules/esl-base-element/core';
+
+import {TraversingQuery} from '../../../src/modules/all';
+
+export class ESLDemoRTLSwitcher extends ESLBaseElement {
+  static is = 'esl-d-rtl-switcher';
+
+  protected _$target: any;
+
+  @boolAttr({readonly: true}) public active: boolean;
+
+  @attr({defaultValue: '::next'}) public target: string;
+
+  @ready
+  protected connectedCallback() {
+    super.connectedCallback();
+    this.updateTargetFromSelector();
+    this.setInitialDirection();
+  }
+
+  @ready
+  protected disconnectedCallback() {
+    super.disconnectedCallback();
+  }
+
+  public get $target(): any {
+    return this._$target;
+  }
+
+  public get $active(): any {
+    return this.$target?.dir === 'rtl';
+  }
+
+  public set $target(newPopupInstance) {
+    this._$target = newPopupInstance;
+    this.bindEvents();
+  }
+
+  protected updateTargetFromSelector() {
+    if (!this.target) return;
+    this.$target = TraversingQuery.first(this.target, this);
+  }
+
+  protected bindEvents() {
+    this.addEventListener('click', this._onClick);
+  }
+
+  protected unbindEvents() {
+    this.removeEventListener('click', this._onClick);
+  }
+
+  protected setInitialDirection() {
+    this.setLTRDirection();
+  }
+
+  public toggleDirection(state: boolean = !this.active) {
+    state ? this.setRTLDirection() : this.setLTRDirection();
+  }
+
+  public setLTRDirection() {
+    this.$target.setAttribute('dir', 'ltr');
+  }
+
+  public setRTLDirection() {
+    this.$target.setAttribute('dir', 'rtl');
+  }
```

Comment:
Can be simplified there is no reason to have so many sub-methods for a simple action.

Reply:
You can convert it to a single method with a boolean term isRTLTargetDirection as a base